tools/src/file_format/config.rs

```diff
@@ -79,6 +79,26 @@ pub struct TreeConfigPaths {
     /// Manually allocated port number to host the livegrep server on, starting
     /// from 8081 why not.
     pub codesearch_port: u32,
+    /// Definitions of SCIP-based indexes to ingest.  Currently it's expected
+    /// that the build script will handle downloading or generating the indexes.
+    #[serde(default)]
+    pub scip_subtrees: BTreeMap<String, ScipSubtreeConfig>,
+}
+
+#[derive(Clone, Debug, Serialize, Deserialize)]
+pub struct ScipSubtreeConfig {
+    /// The path the SCIP index can be found at.
+    pub scip_index_path: String,
```

Comment:
I'm not sure if we have a precedent for this kind of indexer info inside the tree config. It just feels like something that should probably be a build output perhaps? Anyways not opposed to it, it works.

Reply:
Yeah, I'm not 100% on this either.  I think a lot of my decision here was just "having these defined in JSON could be handy and this is a JSON file we already have and use and where we already do the environment variable translations".  A per-tree JSON file or even something like an /etc-style `scip-trees.d` structure where we allow the build system to just dump JSON blobs in a directory and then we union them up together with jq might work much better.  (And the directory could also be pre-populated from the tree's config dir for statically known things.)

I removed it from the stack because we're not using scip-typescript, but I guess I already have a motivating use case for this in the hunk I removed:

```bash
# Find all the JS islands identified by "package.json"; if they have a jsconfig.json
# we would not want to `--infer-tsconfig`, but currently these do not exist in the
# m-c tree, so we don't need to care.
#
# Our find command prints out just the dirname where we found the "package.json",
# which will be "." for the root and which we will want to ignore, so we use grep
# for that.  (Specifically, the package.json at the root of m-c is useless for
# our purposes, it just expresses devDependencies and we do not want
# typescript to treat the entire tree as a single JS project root because it
# categorically is not.)
JS_SCIP_ROOT=$INDEX_ROOT/js-scips
SCRATCH_CONFIG_FILE=$WORKING/config-scratch.json
mkdir -p $JS_SCIP_ROOT
pushd $GIT_ROOT
find * -name package.json -printf '%h\n' | grep -v '^\.$' | while read -r dir; do
    # Use the path as the basis for our name for this index.  "foo/bar/baz" will
    # become "foo__bar__baz".
    SCIP_KEY=${dir//\//__}
    SCIP_FILE=${JS_SCIP_ROOT}/${SCIP_KEY}.scip
    pushd $dir
    # Run the indexer and add an entry to our config file if it succeeds.
    rtx exec nodejs@18 -- scip-typescript index --output ${SCIP_FILE} --infer-tsconfig && \
      jq ".trees[\"mozilla-central\"][\"scip_subtrees\"][\"$SCIP_KEY\"] = { \"scip_index_path\": \"$SCIP_FILE\", \"subtree_root\": \"$dir\"}" $CONFIG_FILE > $SCRATCH_CONFIG_FILE && \
      mv $SCRATCH_CONFIG_FILE $CONFIG_FILE || true
    popd
done
popd
```

The horrible bash there is iteratively updating the environment-evaluated config file we store in the index root in order to cram additional entries into `scip_subtrees` iteratively.


The initial rationale for introducing this mechanism (and before the above bash nightmare):
1. We use the name of the key for this record in `scip_subtrees` above as a human chosen identifier that we include in the symbols we generate to concisely namespace them.
2. I started out without this and it felt like it was going to result in a lot of extra bash hassles in terms of propagating the values around and argument passing, and that there was potential where we might have scip-indexer directly understand what to do from the config file.  I didn't move forward on scip-indexer doing that because the existing per-platform m-c dataflow really favors the existing approach.

Reply:
I'd definitely be interested if you think an `outputs.d` mechanism would make sense and/or what other data you might see us including in such a self-descriptive build byproducts mechanism.

That said, I think I might favor just leaving this setup as-is for this landing with the understand we will replace it.  Noting that in this stack I somehow broke the indexing of `python/` in the tree that works on asuth.searchfox.org, so maybe I can change it up while addressing that.

Reply:
Yeah, leaving as is for now sounds fine to me.

Some more descriptive build output kind of thing sounds worth it. But we'd need to teach each build script to generate it...

Reply:
> Some more descriptive build output kind of thing sounds worth it. But we'd need to teach each build script to generate it...

We could provide wrapper scripts or a wrapper-driver for `scip-typescript`, etc. that help.

That said, the current configs just approximate the same information that could be found by naming the file `SCIP_SUBTREE_NAME.scip` and placing it at the ~root path of where we indexed which provides us with `subtree_root`.  The most natural place for those files to end up is unfortunately in the source tree (and I think the scip-foo binaries already like to do this and can get mad if you try and point the output elsewhere), and we don't zip that up because we already know the source tree contents, so we could also have a `find` script over the source tree for `*.scip` that derives the info for us (assuming that the index names are already appropriate).

But this seems like something we can iterate on based on as part of @nicolas-guichard's work.